app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt

```diff
@@ -0,0 +1,148 @@
+package com.example.abha_create_verify_android.utils
+
+import android.util.Log
+import org.xml.sax.InputSource
+import org.xmlpull.v1.XmlPullParser
+import org.xmlpull.v1.XmlPullParserException
+import org.xmlpull.v1.XmlPullParserFactory
+import java.io.IOException
+import java.io.StringReader
+import javax.xml.parsers.DocumentBuilderFactory
+
+
+class AadhaarXmlAndTextQr(scanData: String?) {
+    private var scannedAadhaarCardInfo: AadhaarCardInfo = AadhaarCardInfo()
+
+    init {
+        if(isXmlData(scanData)) {
+            processXmlData(scanData)
```

Comment:
Can we handle a scenario where something random is scanned which is not aadhaar?